src/Support/SupportRESTController.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * REST API support for Support.
+ *
+ * @package amp-wp
+ */
+
+namespace AmpProject\AmpWP\Support;
+
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AmpProject\AmpWP\Services;
+use WP_REST_Server;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_HTTP_Response;
+use WP_Error;
+
+/**
+ * Class SupportRESTController
+ * REST API support to send AMP support data.
+ *
+ * @package AmpProject\AmpWP\Support
+ */
+class SupportRESTController extends WP_REST_Controller implements Delayed, Service, Registerable {
+
+	/**
+	 * Namespace for REST API endpoint.
+	 *
+	 * @var string $namespace
+	 */
+	public $namespace = 'amp/v1';
+
+	/**
+	 * SupportData instance.
+	 *
+	 * @var SupportData
+	 */
+	public $support_data;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param SupportData $support_data An instance of the SupportData service.
+	 */
+	public function __construct( SupportData $support_data ) {
+
+		$this->support_data = $support_data;
+	}
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+
+		return 'rest_api_init';
+	}
+
+	/**
+	 * Registers all routes for the controller.
+	 */
+	public function register() {
+
+		register_rest_route(
+			$this->namespace,
+			'/send-diagnostic',
+			[
+				[
+					'methods'             => WP_REST_Server::CREATABLE,
+					'args'                => [],
+					'permission_callback' => [ $this, 'permission_callback' ],
+					'callback'            => [ $this, 'callback' ],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Permission check to send support data.
+	 *
+	 * @return bool True if user have permission. Otherwise False.
+	 */
+	public function permission_callback() {
+
+		return current_user_can( 'manage_options' );
+	}
+
+	/**
+	 * Send AMP support data to insight server.
+	 *
+	 * @param WP_REST_Request $request REST API request.
+	 *
+	 * @return WP_Error|WP_HTTP_Response|WP_REST_Response REST API response.
+	 */
+	public function callback( WP_REST_Request $request ) {
+
+		$request_args = $request->get_param( 'args' );
+		$request_args = ( ! empty( $request_args ) && is_array( $request_args ) ) ? $request_args : [];
+
+		$this->support_data->set_args( $request_args );
+		$support_response = $this->support_data->send_data();
```

Comment:
See other note in this PR how it may make sense to use the `injector` to instantiate the `SupportData` here and pass it the args in the constructor.